pkg/container/qemu_runner.go

```diff
@@ -1117,25 +1131,50 @@ func convertHumanToKB(memory string) (int64, error) {
 	}
 
 	// Return the value in kilobytes
-	return num * multiplier / 1024, nil
+	div := KB
+	if strings.Contains(unit, "i") {
+		div = KiB
+	}
+	return num * multiplier / div, nil
 }
 
 func getAvailableMemoryKB() int {
 	mem := 16000000
 
-	f, e := os.Open("/proc/meminfo")
-	if e != nil {
-		return mem
-	}
-	defer f.Close()
+	switch runtime.GOOS {
+	case "linux":
+		f, e := os.Open("/proc/meminfo")
+		if e != nil {
+			return mem
+		}
+		defer f.Close()
 
-	s := bufio.NewScanner(f)
+		s := bufio.NewScanner(f)
 
-	for s.Scan() {
-		var n int
-		if nItems, _ := fmt.Sscanf(s.Text(), "MemTotal: %d kB", &n); nItems == 1 {
-			return n
+		for s.Scan() {
+			var n int
+			if nItems, _ := fmt.Sscanf(s.Text(), "MemTotal: %d kB", &n); nItems == 1 {
+				return n
+			}
+		}
+	case "darwin":
+		cmd := exec.Command("sysctl", "hw.memsize")
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			return mem
+		}
+		outputStr := strings.TrimSpace(string(output))
+		parts := strings.Split(outputStr, ": ")
+		if len(parts) != 2 {
+			return mem
+		}
+
+		memsize, err := strconv.ParseInt(parts[1], 10, 64)
+		if err != nil {
+			return mem
 		}
+		// hw.memsize returns the memory size in bytes
+		return int(memsize / 1024)
```

Comment:
You're using 1000 based units for conversion but here we do 1024, let's change this?

Reply:
Funnily enough, using 1000 causes QEMU to throw exceptions. I don't know why that is.

Reply:
I can revert to using 1024 everywhere.